cache.go

```diff
@@ -193,6 +193,23 @@ func (cache *Cache) Remove(key interface{}) bool {
 	return false
 }
 
+// EvictOldest removes the oldest item from the cache, while also invoking any
+// eviction callback. A bool is returned indicating whether or not an item was
+// removed
+func (cache *Cache) EvictOldest() bool {
+	element := cache.evictionList.Back()
+	if element == nil {
+		return false
+	}
+
+	cache.evictions++
```

Comment:
would this operation need to take place within a mutex?

Reply:
Good catch, it was previously wrapped in one